electionleaflets/apps/leaflets/managers.py

```diff
@@ -0,0 +1,161 @@
+import re
+import unicodedata
+
+from django.contrib.postgres.search import SearchQuery, SearchRank
+from django.db import connection, models
+from django.db.models import F
+
+POPULATE_NAME_SEARCH_COLUMN_SQL = """
```

Comment:
I found this a bit tricky to parse. Having read over this, it looks like there's 2 key parts to it.

One is we're assembling an array of names by pulling any names out of the `people` object and also shoving in the `ynr_person_name` if it exists.

And then the other is we're splitting the name up into parts and weighting them.

It would probably be a bit easier to understand/maintain this in future if we were to add some `--sql-comments` in here explaining what is going on a bit at each stage and why.